haystack/modeling/conversion/transformers.py

```diff
@@ -46,7 +47,7 @@ def convert_to_transformers(adaptive_model):
     @staticmethod
     def convert_from_transformers(
         model_name_or_path,
-        device,
+        device: torch.device,
```

Comment:
`convert_from_transformers` here and for example in `AdaptiveModel` is a method that users would call directly so I'd say that's user-facing. I'm okay with defining it as `torch.device` but supporting `str` as well would simplify the usage for users. It's similar for `load()` of `AdaptiveModel`.
The advantage of the changes as they are in this PR is that it introduces a clear cut where `str` is still supported for `device`: in all components that can be used in a pipeline.

Reply:
To be clear: I approve `device: torch.device`.

Reply:
Ok! I was not too sure if this was to be considered user-facing. Thanks for the confirmation :slightly_smiling_face: